modules/Sources/Dependencies/ZcashSDKEnvironment/ZcashSDKEnvironmentInterface.swift

```diff
@@ -16,6 +18,85 @@ extension DependencyValues {
 }
 
 extension ZcashSDKEnvironment {
+    public enum Servers: String, CaseIterable, Equatable {
+        public enum Constants {
+            public static let udServerKey = "zashi_udServerKey"
+            public static let udCustomServerKey = "zashi_udCustomServerKey"
+        }
+        
+        case mainnet
+        case naNW
+        case saNW
+        case euNW
+        case aiNW
+        case custom
+        
+        public func server() -> String {
+            switch self {
+            case .mainnet: return "mainnet.lightwalletd.com:9067"
+            case .naNW: return "na.lightwalletd.com:443"
+            case .saNW: return "sa.lightwalletd.com:443"
+            case .euNW: return "eu.lightwalletd.com:443"
+            case .aiNW: return "ai.lightwalletd.com:443"
+            case .custom: return "custom"
+            }
+        }
+        
+        public func lightWalletEndpoint(_ userDefaults: UserDefaultsClient) -> LightWalletEndpoint? {
+            switch self {
+            case .mainnet:
+                return LightWalletEndpoint(
+                    address: "mainnet.lightwalletd.com",
+                    port: 9067,
+                    secure: true,
+                    streamingCallTimeoutInMillis: ZcashSDKConstants.streamingCallTimeoutInMillis
+                )
+            case .naNW, .saNW, .euNW, .aiNW:
+                return LightWalletEndpoint(
+                    address: String(self.server().dropLast(4)),
+                    port: 443,
+                    secure: true,
+                    streamingCallTimeoutInMillis: ZcashSDKConstants.streamingCallTimeoutInMillis
+                )
+            case .custom:
+                let udKey = ZcashSDKEnvironment.Servers.Constants.udCustomServerKey
+                if let storedCustomServer = userDefaults.objectForKey(udKey) as? String{
+                    // remove http:// or https:// from the input if present
+                    var input = storedCustomServer
+                    
+                    if input.contains("https://") {
+                        input = String(input.dropFirst(8))
+                    } else if input.contains("http://") {
+                        input = String(input.dropFirst(7))
```

Comment:
Nit: if these strings were constants, then the `dropFirst` could use the length of the constant instead of a separate constant that needs to be kept in sync.

Reply:
```swift
let http = "http://"
                    let https = "https://"
                    if input.contains(https) {
                        input = String(input.dropFirst(https.count))
                    } else if input.contains(http) {
                        input = String(input.dropFirst(http.count))
                    }
```